libs/output-mapping/src/Writer/Table/TableDefinitionColumn.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\OutputMapping\Writer\Table;
+
+use Keboola\Datatype\Definition\Common;
+use Keboola\Datatype\Definition\DefinitionInterface;
+use Keboola\Datatype\Definition\Snowflake;
+
+class TableDefinitionColumn
+{
+
+    private string $name;
+
+    private ?DefinitionInterface $dataTypeDefinition = null;
+
+    private string $baseType;
+
+    public function __construct(string $name, array $metadata, string $nativeTypeClass = null)
+    {
+        $this->name = $name;
+        if ($nativeTypeClass) {
+            $type = null;
+            $options = [];
+            foreach ($metadata as $metadatum) {
+                switch ($metadatum['key']) {
+                    case Common::KBC_METADATA_KEY_TYPE:
+                        $type = $metadatum['value'];
+                        break;
+                    case Common::KBC_METADATA_KEY_LENGTH:
+                        $options['length'] = $metadatum['value'];
+                        break;
+                    case Common::KBC_METADATA_KEY_DEFAULT:
+                        $options['default'] = $metadatum['value'];
+                        break;
+                    case Common::KBC_METADATA_KEY_NULLABLE:
+                        $options['nullable'] = $metadatum['value'];
+                        break;
+                    default:
+                        break;
+                }
+            }
+            if ($type) {
+                $this->dataTypeDefinition = new $nativeTypeClass($type, $options);
+            }
+        }
+        foreach ($metadata as $metadatum) {
+            if ($metadatum['key'] === Common::KBC_METADATA_KEY_BASETYPE) {
+                $this->baseType = $metadatum['value'];
+            }
+        }
+    }
```

Comment:
@odinuv @joker806 I don't like what I've put here so wanted to ping you guys to see if you had maybe some alternative idea of how we can do this :)